grafana-plugin/src/pages/incidents/Incidents.tsx

```diff
@@ -591,6 +629,29 @@ class Incidents extends React.Component<IncidentsPageProps, IncidentsPageState>
     );
   }
 
+  getApplyLabelFilterClickHandler = (label: LabelKeyValue) => {
+    const {
+      store: { filtersStore },
+    } = this.props;
+
+    return () => {
+      const {
+        filters: { label: oldLabelFilter = [] },
+      } = this.state;
+
+      const labelToAddString = `${label.key.id}:${label.value.id}`;
+      if (oldLabelFilter.some((label) => label === labelToAddString)) {
+        return;
+      }
+
+      const newLabelFilter = [...oldLabelFilter, labelToAddString];
+
+      LocationHelper.update({ label: newLabelFilter }, 'partial');
+
+      filtersStore.needToParseFilters = true;
```

Comment:
Let's pls avoid changing observable without an action. Newer versions of MobX will log a warning in every case like that. Using actions have also some other benefits:
![image](https://github.com/grafana/oncall/assets/12073649/f35cfdb3-d078-44ac-8e8d-1f83955131fe)